blocks/api/raw-handling/utils.js

```diff
@@ -63,8 +73,18 @@ export function isAttributeWhitelisted( tag, attribute ) {
 	);
 }
 
-export function isInline( node ) {
-	return !! inlineWhitelist[ node.nodeName.toLowerCase() ];
+function nodeIsInlineOnSpecificTag( nodeName, tagName ) {
+	if ( ! tagName || ! nodeName ) {
+		return false;
+	}
+	return includes(
+		find( inlineWhitelistTagGroups, group => includes( group, tagName.toUpperCase() ) ),
+		nodeName,
+	);
```

Comment:
Using `find` poses a problem if the same tag is found in multiple groups. Example:

```js
const groups = [
  [ 'a', 'b', 'c' ],
  [ 'c', 'd', 'e' ],
];

isInline( 'd', 'c' ); // false, but should be true
```

Furthermore, the time complexity is a lot greater than necessary because of the nested `includes > find > includes` algorithm, when actually **O(n)** is feasible.

Using `Array#some + lodash#includes` helps us reduce complexity and IMO improves readability, though `Array#some + for` would be the fastest, but also much more verbose. See [this working jsfiddle](https://jsfiddle.net/mcsf/88e08w2j/).

Reply:
Thank you for providing a fiddle 👍 And finding the logic bug, as before we used find to get an array and then passed the array I ported that logic but it is now not fitted. I used the some + includes as it is less verbose, and although less efficient we will not have many groups so that should not be noticeable.